client/src/components/Typewriter.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from 'react';
+
+type TypewriterProps = {
+  conversation: string[];
+  onMessageUpdate: (message: string) => void;  // New prop to handle currentMessage updates
+};
+
+const Typewriter: React.FC<TypewriterProps> = ({ conversation, onMessageUpdate }) => {
+  const [currentMessage, setCurrentMessage] = useState('');
+  const [messageIndex, setMessageIndex] = useState(0);
+  const [charIndex, setCharIndex] = useState(0);
+
+  useEffect(() => {
+    if (messageIndex < conversation.length) {
+      if (charIndex < conversation[messageIndex].length) {
+        const timeoutId = setTimeout(() => {
+          const newMessage = currentMessage + conversation[messageIndex][charIndex];
+          setCurrentMessage(newMessage);
+          onMessageUpdate(newMessage);  
+          setCharIndex((prev) => prev + 1);
+        }, 50); // Adjust speed here
```

Comment:
So there's no logic here reading the `character_delay_ms` and `do_type_message` fields, right? That'll be a later problem?

Reply:
yep, later problem

Reply:
Sounds good